wg-api/best-practices.md

```diff
@@ -1,4 +1,168 @@
 # Public API Best Practices
 
-This is a placeholder for a document being drafted by @zcbenz.
-It will be added in a separate PR.
+This is a draft for best practices when designing public APIs.
+
+## Module
+
+### Module Name
+
+If a module exports only one class, the module's name should be the class's name.
+
+For example:
+
+```javascript
+const { BrowserWindow } = require('electron')
+```
+
+If a module includes methods or more than one class, the module's name should be in `lowerCamelCase`.
+
+For example:
+
+```javascript
+const { nativeImage } = require('electron')
+nativeImage.createEmpty()
+```
+
+## Class
+
+### Creating Instance
+
+There are 2 styles for creating instances of classes:
+
+```javascript
+const { BrowserWindow, nativeImage } = require('electron')
+
+// Constructor
+new BrowserWindow(options)
+
+// Factory method
+nativeImage.createFromPath('/path')
+```
+
+#### Constructor
+
+If the class has inheritance relationship with other public classes, constructor should be used.
+
+For example:
+
+```javascript
+const { BrowserWindow, TopLevelWindow } = require('electron')
+const win = new BrowserWindow(options)
+```
+
+This enables users to detect inheritance relationships between classes.
+
+```javascript
+if (win instanceof TopLevelWindow) {
+  // ...
+}
+```
+
+#### Factory method
+
+If the class accepts different kinds of parameters with options, we should consider using factory methods.
```

Comment:
For consistency: `If the class accepts different kinds of parameters with options, consider using factory methods.`